### PR TITLE
templates: update dashboards to include tenant

### PR DESCRIPTION
**Pull request by chloenayon**
Add a tenant variable to the composer dashboard, with the option
to select multiple tenants. Add tenant filter to queries accordingly.

link to dashboard: https://grafana.stage.devshift.net/d/image-builder-worker-with-tenant/image-builder-worker?orgId=1

This pull request includes:

- [ ] adequate testing for the new functionality or fixed issue
- [ ] adequate documentation informing people about the change such as
  - [ ] submit a PR for the [guides](https://github.com/osbuild/guides) repository if this PR changed any behavior described there: https://www.osbuild.org/guides/

### CODE REVIEW

File: templates/dashboards/grafana-dashboard-image-builder-worker-general.configmap.yml [hunk 1]
```diff
@@ -9,2587 +9,2736 @@ metadata:
...
+        "12h",
+        "24h",
+        "2d",
+        "7d",
+        "28d"
+      ]
+    },
+    "timezone": "",
+    "title": "Image Builder Worker with Tenant",
+    "uid": "JLSCLdqnk",
```
Comment (Gundersanne) [inline templates/dashboards/grafana-dashboard-image-builder-worker-general.configmap.yml:2741]: Keep the `image-builder-worker` uid here; it determines the url of the dashboard.

File: templates/dashboards/grafana-dashboard-image-builder-worker-general.configmap.yml [hunk 2]
```diff
@@ -9,2587 +9,2736 @@ metadata:
...
+    "version": 3,
```
Comment (Gundersanne) [inline templates/dashboards/grafana-dashboard-image-builder-worker-general.configmap.yml:2742]: Let's make it version 7?

File: templates/dashboards/grafana-dashboard-image-builder-worker-general.configmap.yml [hunk 3]
```diff
@@ -9,2587 +9,2736 @@ metadata:
...
+    "weekStart": ""
+  }
```
Comment (Gundersanne) [inline templates/dashboards/grafana-dashboard-image-builder-worker-general.configmap.yml:2744]: Newline at the end please ^_^

File: templates/dashboards/grafana-dashboard-image-builder-worker-general.configmap.yml [hunk 4]
```diff
@@ -9,2587 +9,2736 @@ metadata:
...
-      },
-      "editable": true,
-      "fiscalYearStartMonth": 0,
-      "graphTooltip": 0,
-      "id": 33,
-      "iteration": 1652780554062,
-      "links": [],
-      "liveNow": false,
-      "panels": [
+    "annotations": {
```
Comment (kingsleyzissou) [inline templates/dashboards/grafana-dashboard-image-builder-worker-general.configmap.yml:12]: I think we need two more spaces of indentation here

File: templates/dashboards/grafana-dashboard-image-builder-worker-general.configmap.yml [hunk 5]
```diff
@@ -2590,6 +2749,7 @@ data:
       "timezone": "",
       "title": "Image Builder Worker",
       "uid": "image-builder-worker",
-      "version": 6,
+      "version": 7,
       "weekStart": ""
     }
+
```
Comment (Gundersanne) [inline templates/dashboards/grafana-dashboard-image-builder-worker-general.configmap.yml:2755]: nitpick, empty whitespace

File: templates/dashboards/grafana-dashboard-image-builder-worker-general.configmap.yml [hunk 6]
```diff
@@ -2555,11 +2673,52 @@ data:
...
+            "query": "org-15842261, org-15877963, org-15885990, org-16057323",
+            "queryValue": "",
+            "skipUrlSync": false,
+            "type": "custom"
           }
         ]
       },
       "time": {
-        "from": "now-28d",
+        "from": "now-30d",
```
Comment (Gundersanne) [inline templates/dashboards/grafana-dashboard-image-builder-worker-general.configmap.yml:2721]: Keep the now-28d as default; all our SLOs are calculated across 28d and this way the rest of the graphs (which are time-dependent) show the same time period by default; it's a small thing but still.

File: templates/dashboards/grafana-dashboard-image-builder-worker-general.configmap.yml [hunk 7]
```diff
@@ -2487,8 +2605,8 @@ data:
             "auto_min": "10s",
             "current": {
               "selected": false,
-              "text": "6h",
-              "value": "6h"
+              "text": "7d",
+              "value": "7d"
```
Comment (Gundersanne) [inline templates/dashboards/grafana-dashboard-image-builder-worker-general.configmap.yml:2609]: keep this at 6h by default, this determines the granularity/the amount of the data points shown; 6h is a happy medium between smoothing out and showing spikes.

File: templates/dashboards/grafana-dashboard-image-builder-worker-general.configmap.yml [hunk 8]
```diff
@@ -2465,9 +2583,9 @@ data:
         "list": [
           {
             "current": {
-              "selected": true,
-              "text": "app-sre-prod-04-prometheus",
-              "value": "app-sre-prod-04-prometheus"
+              "selected": false,
+              "text": "app-sre-stage-01-prometheus",
+              "value": "app-sre-stage-01-prometheus"
```
Comment (Gundersanne) [inline templates/dashboards/grafana-dashboard-image-builder-worker-general.configmap.yml:2588]: Default should be app-sre-prod-04-prometheus here so also revert this.

File: templates/dashboards/grafana-dashboard-image-builder-worker-general.configmap.yml [hunk 9]
```diff
@@ -136,8 +137,12 @@ data:
           "pluginVersion": "8.5.2",
           "targets": [
             {
+              "datasource": {
+                "type": "prometheus",
+                "uid": "gH4yqWBnz"
+              },
```
Comment (Gundersanne) [inline templates/dashboards/grafana-dashboard-image-builder-worker-general.configmap.yml:143]: don't pin the datasource; it should be "uid": "${datasource}" always.

File: templates/dashboards/grafana-dashboard-image-builder-worker-general.configmap.yml [hunk 10]
```diff
@@ -326,8 +341,12 @@ data:
           },
           "targets": [
             {
+              "datasource": {
+                "type": "prometheus",
+                "uid": "gH4yqWBnz"
```
Comment (Gundersanne) [inline templates/dashboards/grafana-dashboard-image-builder-worker-general.configmap.yml:346]: Same

File: templates/dashboards/grafana-dashboard-image-builder-worker-general.configmap.yml [hunk 11]
```diff
@@ -520,8 +544,12 @@ data:
           "pluginVersion": "8.1.5",
           "targets": [
             {
+              "datasource": {
+                "type": "prometheus",
+                "uid": "gH4yqWBnz"
```
Comment (Gundersanne) [inline templates/dashboards/grafana-dashboard-image-builder-worker-general.configmap.yml:549]: Same

File: templates/dashboards/grafana-dashboard-image-builder-worker-general.configmap.yml [hunk 12]
```diff
@@ -761,26 +795,38 @@ data:
...
               "hide": false,
               "interval": "",
               "legendFormat": "p95",
               "range": true,
               "refId": "B"
             },
             {
+              "datasource": {
+                "type": "prometheus",
+                "uid": "gH4yqWBnz"
```
Comment (Gundersanne) [inline templates/dashboards/grafana-dashboard-image-builder-worker-general.configmap.yml:826]: idem

File: templates/dashboards/grafana-dashboard-image-builder-worker-general.configmap.yml [hunk 13]
```diff
@@ -962,8 +1013,12 @@ data:
           "pluginVersion": "8.5.2",
           "targets": [
             {
+              "datasource": {
+                "type": "prometheus",
+                "uid": "gH4yqWBnz"
```
Comment (Gundersanne) [inline templates/dashboards/grafana-dashboard-image-builder-worker-general.configmap.yml:1018]: "${datasource}"

File: templates/dashboards/grafana-dashboard-image-builder-worker-general.configmap.yml [hunk 14]
```diff
@@ -1060,8 +1116,12 @@ data:
           "pluginVersion": "8.1.5",
           "targets": [
             {
+              "datasource": {
+                "type": "prometheus",
+                "uid": "gH4yqWBnz"
```
Comment (Gundersanne) [inline templates/dashboards/grafana-dashboard-image-builder-worker-general.configmap.yml:1121]: etc.

**Review by Gundersanne — CHANGES_REQUESTED** [review]
Good direction. A few things

- don't pin of the datasource ( I made that mistake a few times :|)
- the job wait duration on the db you linked is collapsed by default, it can be expanded
- selecting multiple tenants doesn't work, instead of a multi select we could just have a "*" option (and "*" can be the default)


A lot of this is stuff that gets changed randomly during exporting (like the datasource thing). `git add -p` is handy here. And it's worthwhile to reimport the json after having exported/committed it to see the diff between them.

File: templates/dashboards/grafana-dashboard-image-builder-worker-general.configmap.yml [hunk 15]
```diff
@@ -234,16 +240,24 @@ data:
...
+              "expr": "(sum(rate(image_builder_worker_total_jobs{type=~\"osbuild:.*\", status!=\"4xx\", tenant=\"$tenant\"}[$interval])) OR on() vector(0))\n- \n(sum(rate(image_builder_worker_total_jobs{type=~\"osbuild:.*\", status=\"5xx\", tenant=\"$tenant\"}[$interval])) OR on() vector(0))",
               "hide": false,
               "interval": "",
               "legendFormat": "success/sec",
               "refId": "A"
             },
             {
+              "datasource": {
+                "type": "prometheus",
+                "uid": ""${datasource}""
```
Comment (kingsleyzissou) [inline templates/dashboards/grafana-dashboard-image-builder-worker-general.configmap.yml:257]: Same here. I think there are a few like this

File: templates/dashboards/grafana-dashboard-image-builder-worker-general.configmap.yml [hunk 16]
```diff
@@ -234,16 +240,24 @@ data:
           },
           "targets": [
             {
+              "datasource": {
+                "type": "prometheus",
+                "uid": ""${datasource}""
+              },
               "exemplar": true,
-              "expr": "(sum(rate(image_builder_worker_total_jobs{type=~\"osbuild:.*\", status!=\"4xx\"}[$interval])) OR on() vector(0))\n- \n(sum(rate(image_builder_worker_total_jobs{type=~\"osbuild:.*\", status=\"5xx\"}[$interval])) OR on() vector(0))",
+              "expr": "(sum(rate(image_builder_worker_total_jobs{type=~\"osbuild:.*\", status!=\"4xx\", tenant=\"$tenant\"}[$interval])) OR on() vector(0))\n- \n(sum(rate(image_builder_worker_total_jobs{type=~\"osbuild:.*\", status=\"5xx\", tenant=\"$tenant\"}[$interval])) OR on() vector(0))",
```
Comment (kingsleyzissou) [inline templates/dashboards/grafana-dashboard-image-builder-worker-general.configmap.yml:248]: If you change this `tenant=\"$tenant\"` to `tenant=~\"$tenant\"` for all instances, the selection should work.

File: templates/dashboards/grafana-dashboard-image-builder-worker-general.configmap.yml [hunk 17]
```diff
@@ -2509,7 +2627,7 @@ data:
                 "value": "1h"
               },
               {
-                "selected": true,
+                "selected": false,
```
Comment (kingsleyzissou) [inline templates/dashboards/grafana-dashboard-image-builder-worker-general.configmap.yml:2630]: We should leave `6h` as the default resolution

**Review by kingsleyzissou — CHANGES_REQUESTED** [review]

**Comment by kingsleyzissou** [comment]
Sorry been meaning to get back to this, I had a look this morning and changing from `tenant=\"$tentant\"` to `tenant=~\"$tentant\"` should do the trick.

**Review by kingsleyzissou — APPROVED** [review]
Thanks!

File: templates/dashboards/grafana-dashboard-image-builder-worker-general.configmap.yml [hunk 18]
```diff
@@ -2555,6 +2673,47 @@ data:
             "query": "0.95",
             "skipUrlSync": false,
             "type": "constant"
+          },
+          {
+            "current": {
```
Comment (kingsleyzissou) [inline templates/dashboards/grafana-dashboard-image-builder-worker-general.configmap.yml:2678]: hey @chloenayon sorry for the change request on this :/
See this conversation here:
https://github.com/RedHatInsights/image-builder-frontend/pull/753#discussion_r921235240

Could you please update this to have an "All" option and making it the default. I think you can add it through the GUI:
image

**Review by kingsleyzissou — CHANGES_REQUESTED** [review]

File: templates/dashboards/grafana-dashboard-image-builder-worker-general.configmap.yml (hunk 18 above)
Comment (kingsleyzissou) [reply templates/dashboards/grafana-dashboard-image-builder-worker-general.configmap.yml:2678]: Setting the custom all value to `.*` will give us a regex wildcard

**Review by kingsleyzissou — DISMISSED** [review]
Awesome, thanks so much!

File: templates/dashboards/grafana-dashboard-image-builder-worker-general.configmap.yml [hunk 19]
```diff
@@ -2589,7 +2742,7 @@ data:
       },
       "timezone": "",
       "title": "Image Builder Worker",
-      "uid": "image-builder-worker",
-      "version": 6,
+      "uid": "image-builder-worker-with-tenant",
+      "version": 2,
```
Comment (Gundersanne) [inline templates/dashboards/grafana-dashboard-image-builder-worker-general.configmap.yml:2746]:
```
 "uid": "image-builder-worker",
      "version": 7,
```

-> bump version by 1, keep uid the same

**Review by Gundersanne — CHANGES_REQUESTED** [review]
1 last nitpick

**Review by Gundersanne — APPROVED** [review]
lgtm, thank you!

**Comment by Gundersanne** [comment]
dashboards updated, codebase/tests not affect. admin merging

**Pull request merged**